### PR TITLE
Feature/GitHub action  ci

### PR DESCRIPTION
**Pull request by JarrodMFlesch**
Hey @esoubihe here is the PR for the github actions - it took me a little bit to understand how they function, but I have a much better understanding now. So with that being said, I would love for you to give these a test and report back to me with any feedback!
<br />

CI Test will run when:
- PR or Push gets made to Develop
- PR or Push gets made to Master

CI Release will run when:
- PR or Push gets made to Master and the CI Test passes

NOTE: You will need to update the Github Token Secrets to include your NPM_TOKEN, you can see it is commented out at the bottom of the ci.yml file.
<br />
The releases are based on Semantic Releases criteria, i.e. using **feat:** and **fix:** with your commit messages.
Let me know if you run into anything!

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    - name: NPM Install ci
+      run: npm ci
+
+    - name: Publish with Semantic Release
+      run: npx semantic-release
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        # for npm publishing to work, remove: "private: true" in the package.json file and add a "NPM_TOKEN" key to your github secrets
+        # NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```
Comment (esoubihe) [inline .github/workflows/ci.yml:61]: @JarrodMFlesch I think we need to uncomment this in order to be able to push it to NPM.

File: package.json [hunk 2]
```diff
@@ -3,6 +3,7 @@
   "version": "1.1.3",
   "description": "Metal UI Design System",
   "main": "dist/index.js",
+  "private": true,
```
Comment (esoubihe) [inline package.json:6]: I believe this was left here by accident?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (esoubihe) [reply .github/workflows/ci.yml:61]: Sorry, just saw your comment. We already have `NPM_TOKEN` setup.

Comment (esoubihe) [reply .github/workflows/ci.yml:61]: Thanks. Feel free to amend that as I'm unable to push to your PR

**Comment by JarrodMFlesch** [comment]
@esoubihe and team - Does anyone know how we can fix the `package-lock.json` conflict? I had to remove node modules and reinstall to get my workflow scripts to run properly, so they are not in sync anymore. When I pull from master I don't get a conflict, so I am unable to resolve it locally, any ideas?

**Comment by nathanmrtns** [comment]
> @esoubihe and team - Does anyone know how we can fix the `package-lock.json` conflict? I had to remove node modules and reinstall to get my workflow scripts to run properly, so they are not in sync anymore. When I pull from master I don't get a conflict, so I am unable to resolve it locally, any ideas?

Did you try deleting your local package-lock file, regenerating and pull from master to see if you get any conflicts?

**Comment by esoubihe** [comment]
`package-lock.json` exists for a reason. We should avoid deleting it. 

It should be ok this once based on what I've seen for the current dependencies, but it's best to adjust your workflow to keep package and lock updated than it is to take the risk of unknowingly updating dependencies that could break things.

**Comment by JarrodMFlesch** [comment]
@nathanmrtns I will give that a shot!

**Comment by JarrodMFlesch** [comment]
@esoubihe & @nathanmrtns So I think I have gotten to the bottom of this. I believe the lock file on master is actually _out of date_, if you run `npm ci` (which is what I am running inside my ci) you will see that the lock file is out of date compared to the actual package versions in the package.json. 

Can we have this fixed on master to fix this conflict? That is the place this issue should be resolved.

**Comment by nathanmrtns** [comment]
@JarrodMFlesch I will fork your repository to see if I can fix it

**Comment by nathanmrtns** [comment]
Hey @JarrodMFlesch can you check if merging[ THIS PR ](https://github.com/trouble/metal-ui/pull/11)into your branch fixes the problem?

**Comment by JarrodMFlesch** [comment]
@nathanmrtns yep that seemed to fix the package lock conflict 👍

**Comment by nathanmrtns** [comment]
> @nathanmrtns This is used to fix an issue with invalid blobs in storybook.

I didn't know, I just wanted to change the package lock in this pr to this branch

File: .storybook/main.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 module.exports = {
-  stories: ["../src/**/*.stories.@(js|mdx)"],
+  stories: ["../src/**/*.stories.(js|mdx)"],
```
Comment (nathanmrtns) [inline .storybook/main.js:2]:
```suggestion
  stories: ["../src/**/*.stories.@(js|mdx)"],
```

Comment (nathanmrtns) [reply .storybook/main.js:2]: Just to revert that change.

**Review by nathanmrtns — CHANGES_REQUESTED** [review]

File: .storybook/main.js (hunk 3 above)
Comment (JarrodMFlesch) [reply .storybook/main.js]: reverted here c050d8b

**Review by nathanmrtns — APPROVED** [review]

**Pull request merged**